File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/inheritance/Administrator.java
package org.babyfish.jimmer.sql.model.inheritance;

import org.babyfish.jimmer.sql.*;

@Entity
@KeyUniqueConstraint
public interface Administrator extends AdministratorBase, UserInfo {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    long getId();
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/model/inheritance/Administrator.kt
package org.babyfish.jimmer.sql.kt.model.inheritance

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.GeneratedValue
import org.babyfish.jimmer.sql.GenerationType
import org.babyfish.jimmer.sql.Id

@Entity
interface Administrator : AdministratorBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long
}